pingcap/tidb
# ddl,table/tables: handle partition by range column when len(column)=1

### What problem does this PR solve? 

```
create table log_message_1(
add_time datetime not null default '2000-01-01 00:00:00',
log_level int unsigned not null default '0',
log_host varchar(32) not null,
service_name varchar(32) not null,
message varchar(2000)
)partition by range columns(add_time)(
partition p201403 values less than ('2014-04-01'),
partition p201404 values less than ('2014-05-01'),
partition p201405 values less than ('2014-06-01'),
partition p201406 values less than ('2014-07-01'),
partition p201407 values less than ('2014-08-01'),
partition p201408 values less than ('2014-09-01'),
partition p201409 values less than ('2014-10-01'),
partition p201410 values less than ('2014-11-01')
)
```

Fix error:

```
2019/01/15 17:06:59.808 partition.go:128: [error] wrong table partition expression:line 1 column 10 near " < ("2014-04-01"))" (total length 28)
github.com/pingcap/errors.AddStack
        /home/jenkins/workspace/build_tidb_2.1/go/pkg/mod/github.com/pingcap/errors@v0.11.0/errors.go:174
github.com/pingcap/errors.Trace
        /home/jenkins/workspace/build_tidb_2.1/go/pkg/mod/github.com/pingcap/errors@v0.11.0/juju_adaptor.go:12
github.com/pingcap/parser.(*Parser).Parse
        /home/jenkins/workspace/build_tidb_2.1/go/pkg/mod/github.com/pingcap/parser@v0.0.0-20181207085916-6c21d4344dfa/yy_parser.go:111
github.com/pingcap/tidb/expression.ParseSimpleExprWithTableInfo
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/expression/simple_rewriter.go:40
github.com/pingcap/tidb/table/tables.generatePartitionExpr
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/table/tables/partition.go:125
github.com/pingcap/tidb/table/tables.newPartitionedTable
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/table/tables/partition.go:66
github.com/pingcap/tidb/table/tables.TableFromMeta
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/table/tables/tables.go:139
github.com/pingcap/tidb/infoschema.(*Builder).applyCreateTable
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/infoschema/builder.go:178
github.com/pingcap/tidb/infoschema.(*Builder).ApplyDiff
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/infoschema/builder.go:94
github.com/pingcap/tidb/domain.(*Domain).tryLoadSchemaDiffs
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/domain/domain.go:231
github.com/pingcap/tidb/domain.(*Domain).loadInfoSchema
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/domain/domain.go:104
github.com/pingcap/tidb/domain.(*Domain).Reload
        /home/jenkins/workspace/build_tidb_2.1/go/src/github.com/pingcap/tidb/domain/domain.go:313
github.com/pingcap/tidb/domain.(*Domain).loadSchemaInLoop
```

### What is changed and how it works?

Add preliminary support for `partition by range column`
'create table ... partition by range column (c)'
can be transformed to 'create table ... partiton by range c'


### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Side effects

With the introduction of this new partition type, we have to check more things:

1. Check value constraint for `create table ... partition by column`
2. Check value constraint for `alter table add partition ...`
3. Add CRUD test for partition by column(c)
